core/src/main/java/org/apache/accumulo/fate/Fate.java

```diff
@@ -226,11 +223,17 @@ public Fate(T environment, TStore<T> store, Function<Repo<T>,String> toLogStrFun
    * Launches the specified number of worker threads.
    */
   public void startTransactionRunners(AccumuloConfiguration conf) {
-    int numThreads = conf.getCount(Property.MANAGER_FATE_THREADPOOL_SIZE);
-    executor = ThreadPools.createExecutorService(conf, Property.MANAGER_FATE_THREADPOOL_SIZE);
-    for (int i = 0; i < numThreads; i++) {
-      executor.execute(new TransactionRunner());
-    }
+    final ThreadPoolExecutor pool = (ThreadPoolExecutor) ThreadPools.createExecutorService(conf,
+        Property.MANAGER_FATE_THREADPOOL_SIZE);
+    fatePoolWatcher = ThreadPools.createGeneralScheduledExecutorService(conf);
```

Comment:
The Accumulo servers used to have a shared timer pool.  Not sure if that is still a thing, if it is could be used to avoid creating these watcher threads.

Reply:
I looked back in 1.10 for shared SimpleTimer's. Please let me know if you happen to have a location on this as I would be interested to see if I missed something when I did the Threads/ThreadPools work. In this specific case, only 1 scheduled runnable is going to be created as `startTransactionRunners` is only called once in Manager.run().

Reply:
Looking back at 1.10 I See the following which created a singleton timer that was shared in the process with a configurable number of threads.

https://github.com/apache/accumulo/blob/bbe2de8a8c00eb0b7cc01766ac6e6db672bf8444/server/base/src/main/java/org/apache/accumulo/server/util/time/SimpleTimer.java#L60-L71

AFAICT now in 2.1.0 there is no longer a process wide singleton timer related to the config `general.server.simpletimer.threadpool.size`?

Reply:
That was my concern, that maybe I missed that. However, now that the creation of that ThreadPool is singularly located in `ThreadPools.createGeneralScheduledExecutorService`, it should be easy to make it a singleton that is only created if it does not already exist. It's called from 33 locations across the code base.

Reply:
It is used by the TabletServerBatchWriter. I think putting it on the ServerContext would render that unreachable.

Reply:
I might be able to put it on AccumuloConfiguration.

Reply:
> It is used by the TabletServerBatchWriter. I think putting it on the ServerContext would render that unreachable.

May make sense to have it in the client context for client side code.  Server context extends client context, so maybe that would work out nicely.